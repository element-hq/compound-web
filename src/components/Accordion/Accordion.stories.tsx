/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { Meta } from "@storybook/react";
import { Item } from "./Accordion";

import { Root as AccordionComponent } from "./Accordion";

const LOREM_IPSUM =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.";

export default {
  title: "Accordion",
  component: AccordionComponent,
  tags: ["autodocs"],
  argTypes: {
    onValueChange: { action: "onValueChange" },
  },
  args: {
    collapsible: true,
    children: (
      <>
        <Item trigger="Item 1" value="item1">
          {LOREM_IPSUM}
        </Item>
        <Item trigger="Item 2" value="item2">
          {LOREM_IPSUM}
        </Item>
        <Item trigger="Item 3" value="item3">
          {LOREM_IPSUM}
        </Item>
      </>
    ),
  },
} as Meta<typeof AccordionComponent>;

export const Default = {
  args: {},
  parameters: {},
};

export const Single = {
  args: {
    type: "single",
  },
  parameters: {},
};

export const Multiple = {
  args: {
    type: "multiple",
  },
  parameters: {},
};

export const SingleControlled = {
  args: {
    value: "item1",
    type: "single",
  },
  parameters: {},
};

export const MultipleControlled = {
  args: {
    value: ["item1", "item2"],
    type: "multiple",
  },
  parameters: {},
};
