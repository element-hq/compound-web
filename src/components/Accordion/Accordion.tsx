/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classNames from "classnames";
import React, { forwardRef, PropsWithChildren, ReactNode } from "react";
import ChevronRightIcon from "@vector-im/compound-design-tokens/icons/chevron-right.svg";
import * as RadixAccordion from "@radix-ui/react-accordion";

import styles from "./Accordion.module.css";

/**
 * See https://www.radix-ui.com/primitives/docs/components/accordion#root
 */
type AccordionProps = {
  /**
   * The CSS class name.
   */
  className?: string;
} & React.ComponentProps<typeof RadixAccordion.Root>;

/**
 * Thin wrapper around Radix UI Accordion component
 *
 * See radix documentation: https://www.radix-ui.com/primitives/docs/components/accordion
 */
export const Root = forwardRef<HTMLDivElement, AccordionProps>(function Root(
  { children, className, ...props }: PropsWithChildren<AccordionProps>,
  ref,
) {
  // TODO: Update the class name to something related to the component name
  const classes = classNames(styles.root, className);
  return (
    <RadixAccordion.Root {...props} className={classes} ref={ref}>
      {children}
    </RadixAccordion.Root>
  );
});

type TriggerProps = {
  /**
   * The CSS class name.
   */
  className?: string;
} & React.ComponentProps<typeof RadixAccordion.Trigger>;
const Trigger = forwardRef<HTMLButtonElement, TriggerProps>(function Trigger(
  { children, className, ...props }: TriggerProps,
  forwardedRef,
) {
  return (
    <RadixAccordion.Header>
      <RadixAccordion.Trigger
        className={classNames(styles.trigger, className)}
        {...props}
        ref={forwardedRef}
      >
        {children}
        <ChevronRightIcon
          height={24}
          width={24}
          className={styles.icon}
          aria-hidden
        />
      </RadixAccordion.Trigger>
    </RadixAccordion.Header>
  );
});

type ItemProps = {
  /**
   * The CSS class name.
   */
  className?: string;
  /**
   * Unique identifier for this item within the accordion
   */
  value: string;
  /**
   * The element or text used as the header of the accordion
   * Will be wrapped in a button element by Radix
   */
  trigger: ReactNode | string;
} & React.ComponentProps<typeof RadixAccordion.Item>;
export const Item = forwardRef<HTMLDivElement, ItemProps>(function Item(
  {
    value,
    trigger,
    children,
    className,
    ...props
  }: PropsWithChildren<ItemProps>,
  ref,
) {
  const classes = classNames(styles.root, className);
  return (
    <RadixAccordion.Item {...props} className={classes} value={value} ref={ref}>
      <Trigger>{trigger}</Trigger>

      <RadixAccordion.Content className={styles.content}>
        {children}
      </RadixAccordion.Content>
    </RadixAccordion.Item>
  );
});
