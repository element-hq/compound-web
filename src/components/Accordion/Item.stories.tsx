/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { Meta, StoryFn } from "@storybook/react";

import { Text } from "../Typography/Text";
import { Root, Item } from "./Accordion";

const LOREM_IPSUM =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.";

export default {
  title: "Accordion/Item",
  component: Item,
  tags: ["autodocs"],
  argTypes: {},
  args: {
    trigger: (
      <Text as="span" size="md">
        Click me!
      </Text>
    ),
    children: <Text>{LOREM_IPSUM}</Text>,
    value: "item1",
  },
  decorators: [
    (Story: StoryFn) => (
      <Root collapsible>
        <Story />
      </Root>
    ),
  ],
} as Meta<typeof Item>;

export const Default = {
  args: {},
};

export const LongTrigger = {
  args: {
    trigger: (
      <Text as="span" size="md">
        Charles &quot;Charlie&quot; Brown is the principal character of the
        comic strip Peanuts, syndicated in daily and Sunday newspapers in
        numerous countries all over the world.
      </Text>
    ),
  },
};
