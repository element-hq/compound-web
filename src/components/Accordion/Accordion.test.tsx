/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import { describe, it, expect, vi, afterEach } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";

import * as stories from "./Accordion.stories";
import { composeStories, composeStory } from "@storybook/react";

const { Default, Single, Multiple } = composeStories(stories);

describe("<Accordion />", () => {
  it("renders an uncontrolled single accordion by default", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });

  it("renders an uncontrolled single accordion", () => {
    const { container } = render(<Single />);
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText("Item 1"));

    expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
      "open",
    );
    fireEvent.click(screen.getByText("Item 2"));

    expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
      "closed",
    );
    expect(screen.getByText("Item 2").getAttribute("data-state")).toEqual(
      "open",
    );
  });

  it("renders an uncontrolled multiple accordion", () => {
    const { container } = render(<Multiple />);
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText("Item 1"));
    expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
      "open",
    );

    fireEvent.click(screen.getByText("Item 2"));
    // both open
    expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
      "open",
    );
    expect(screen.getByText("Item 2").getAttribute("data-state")).toEqual(
      "open",
    );
  });

  describe("single controlled", () => {
    const onValueChange = vi.fn();
    // compose the story with our mocked onValueChange
    const SingleControlled = composeStory(
      {
        ...stories.SingleControlled,
        args: {
          ...stories.SingleControlled.args,
          onValueChange,
        },
      },
      stories.default,
    );

    afterEach(() => {
      onValueChange.mockClear();
    });

    it("renders a controlled single accordion with an item open", () => {
      render(<SingleControlled />);

      expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
        "open",
      );
    });

    it("handles click on an expanded item", () => {
      render(<SingleControlled />);

      fireEvent.click(screen.getByText("Item 1"));
      expect(onValueChange).toHaveBeenCalledWith("");
    });

    it("handles click on a collapsed item", () => {
      render(<SingleControlled />);

      fireEvent.click(screen.getByText("Item 2"));
      expect(onValueChange).toHaveBeenCalledWith("item2");
    });
  });

  describe("multiple controlled", () => {
    const onValueChange = vi.fn();
    // compose the story with our mocked onValueChange
    const MultipleControlled = composeStory(
      {
        ...stories.MultipleControlled,
        args: {
          ...stories.MultipleControlled.args,
          onValueChange,
        },
      },
      stories.default,
    );

    afterEach(() => {
      onValueChange.mockClear();
    });

    it("renders a controlled single accordion with an item open", () => {
      render(<MultipleControlled />);

      expect(screen.getByText("Item 1").getAttribute("data-state")).toEqual(
        "open",
      );
      expect(screen.getByText("Item 2").getAttribute("data-state")).toEqual(
        "open",
      );
    });

    it("handles click on an expanded item", () => {
      render(<MultipleControlled />);

      fireEvent.click(screen.getByText("Item 1"));
      // item1 removed from array of open items
      expect(onValueChange).toHaveBeenCalledWith(["item2"]);
    });

    it("handles click on a collapsed item", () => {
      render(<MultipleControlled />);

      fireEvent.click(screen.getByText("Item 3"));
      // item3 added to array of open items
      expect(onValueChange).toHaveBeenCalledWith(["item1", "item2", "item3"]);
    });
  });
});
