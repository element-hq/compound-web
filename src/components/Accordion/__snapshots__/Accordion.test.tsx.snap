// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Accordion /> > renders an uncontrolled multiple accordion 1`] = `
<div>
  <div
    class="_root_eebbdc"
    data-orientation="vertical"
  >
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:rd:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rc:"
          type="button"
        >
          Item 1
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rc:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:rd:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:rf:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:re:"
          type="button"
        >
          Item 2
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:re:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:rf:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:rh:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rg:"
          type="button"
        >
          Item 3
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rg:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:rh:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;

exports[`<Accordion /> > renders an uncontrolled single accordion 1`] = `
<div>
  <div
    class="_root_eebbdc"
    data-orientation="vertical"
  >
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r7:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r6:"
          type="button"
        >
          Item 1
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r6:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r7:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r9:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r8:"
          type="button"
        >
          Item 2
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r8:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r9:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:rb:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:ra:"
          type="button"
        >
          Item 3
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:ra:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:rb:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;

exports[`<Accordion /> > renders an uncontrolled single accordion by default 1`] = `
<div>
  <div
    class="_root_eebbdc"
    data-orientation="vertical"
  >
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          Item 1
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          Item 2
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="_root_eebbdc"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="_trigger_eebbdc"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          Item 3
          <svg
            aria-hidden="true"
            class="_icon_eebbdc"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.70005 17.3C8.51672 17.1167 8.42505 16.8834 8.42505 16.6C8.42505 16.3167 8.51672 16.0834 8.70005 15.9L12.6 12L8.70005 8.10005C8.51672 7.91672 8.42505 7.68338 8.42505 7.40005C8.42505 7.11672 8.51672 6.88338 8.70005 6.70005C8.88338 6.51672 9.11672 6.42505 9.40005 6.42505C9.68338 6.42505 9.91672 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8709 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8709 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91672 17.4834 9.68338 17.575 9.40005 17.575C9.11672 17.575 8.88338 17.4834 8.70005 17.3Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="_content_eebbdc"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
